include/deal.II/lac/arpack_solver.h

```diff
@@ -104,43 +121,62 @@ class ArpackSolver : public Subscriptor
     both_ends
   };
 
-  /**
-   * Standardized data struct to pipe additional data to the solver, should it
-   * be needed.
-   */
-  struct AdditionalData
-  {
-    const unsigned int number_of_arnoldi_vectors;
-    const WhichEigenvalues eigenvalue_of_interest;
-    AdditionalData(
-      const unsigned int number_of_arnoldi_vectors = 15,
-      const WhichEigenvalues eigenvalue_of_interest = largest_magnitude);
-  };
 
   /**
    * Access to the object that controls convergence.
    */
   SolverControl &control () const;
 
+
   /**
    * Constructor.
    */
-  ArpackSolver(SolverControl &control,
-               const AdditionalData &data = AdditionalData());
+  ArpackSolver(SolverControl &control);
+
 
   /**
    * Solve the generalized eigensprectrum problem $A x=\lambda B x$ by calling
-   * the <code>dneupd</code> and <code>dnaupd</code> functions of ARPACK.
+   * the <code>dneupd</code> and <code>dnaupd</code> functions of
+   * ARPACK.
+   *
+   * The function returns a vector of eigenvalues of length <i>n</i>
+   * and a vector of eigenvectors, where the latter should be twice
+   * the size of the eigenvalue vector. The first <i>n</i> vectors in
+   * <code>eigenvectors</code> will be the real parts of the
+   * eigenvectors, the second <i>n</i> the imaginary parts.
+   *
+   * @param B The inner product of the underlying space, typically the
+   * mass matrix. Only its function <code>vmult()</code> is used.
+   *
+   * @param inverse This is the (possibly shifted) inverse that is actually
+   * used instead of <code>A</code>. Only its function <code>vmult()</code>
+   * is used.
+   *
+   * @param eigenvalues is a vector of complex numbers in which the
+   * eigenvalues are returned. The size of the vector determines the number
+   * of eigenvalues to be computed.
+   *
+   * @param eigenvectors is a <b>real</b> vector of eigenvectors,
+   * the first half containing the real parts and the second half
+   * containing the corresponding imaginary parts of the eigenvectors.
+   * Therefore, its length should be twice the number of eigenvalues. The
+   * vectors have to be initialized to match the matrices.
+   *
+   * @param eigenvalue_of_interest encodes the eigenvalue of interest given
+   * by the enum WhichEigenvalues
```

Comment:
", where the qualifiers in the enum refer to the eigenvalues of the shifted inverse."